lib/time/day_to_str.h

```diff
@@ -17,32 +16,36 @@
 #include "string/strtcpy.h"
 
 
-#define DAY_TO_STR(str, day)   date_to_str(NITEMS(str), str, day * DAY)
+#define DAY_TO_STR(str, day)   day_to_str(NITEMS(str), str, day)
 
 
-inline void date_to_str(size_t size, char buf[size], long date);
+inline void day_to_str(size_t size, char buf[size], long day);
 
 
 inline void
-date_to_str(size_t size, char buf[size], long date)
+day_to_str(size_t size, char buf[size], long day)
 {
-	time_t           t;
+	time_t           date;
 	const struct tm  *tm;
 
-	t = date;
-	if (date < 0) {
-		(void) strtcpy(buf, "never", size);
+	if (day < 0) {
+		strtcpy(buf, "never", size);
 		return;
 	}
 
-	tm = gmtime(&t);
+	if (__builtin_mul_overflow(day, DAY, &date)) {
+		strtcpy(buf, "future", size);
+		return;
+	}
+
+	tm = gmtime(&date);
 	if (tm == NULL) {
-		(void) strtcpy(buf, "future", size);
+		strtcpy(buf, "future", size);
```

Comment:
I realize that there are no callers who could get hit by this, however if buf were too short for "future", then buf would be truncated and not null-terminated.

This seems a bad precedent to set.  It seems like, unless the caller is prepared for this, we should check strtcpy return value and set buf[size-1] = 0 if truncation occurred.

I've merged this pr, my comment here is a more general question.

Reply:
strtcpy() NULL terminates short copies.  It's almost equivalent to the kernel's strscpy(9).

Reply:
```c
$ grepc -tfd strtcpy .
./lib/string/strtcpy.h:inline ssize_t
strtcpy(char *restrict dst, const char *restrict src, size_t dsize)
{
	bool    trunc;
	size_t  dlen, slen;

	if (dsize == 0)
		return -1;

	slen = strnlen(src, dsize);
	trunc = (slen == dsize);
	dlen = slen - trunc;

	stpcpy(mempcpy(dst, src, dlen), "");

	if (trunc)
		return -1;

	return slen;
}
```

The main work is here:
```c
stpcpy(mempcpy(dst, src, dlen), "");
```

And that `stpcpy(..., "");` ensures null termination.

Reply:
Ah, thanks.

I was inverting the value of @trunc in my head.

Looks good.